src/corefx/System.Globalization.Native/casing.cpp

```diff
@@ -13,87 +13,140 @@
 
-    if (bToUpper)
-    {
-        if (bTurkishCasing)
-        {
-            // bToUpper && bTurkishCasing
-            while (srcIdx < cwSrcLength)
-            {
-                // In turkish casing, LATIN SMALL LETTER I (U+0069) upper cases to LATIN
-                // CAPITAL LETTER I WITH DOT ABOVE (U+0130).
-                U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
-                dstCodepoint = ((srcCodepoint == (UChar32)0x0069) ? (UChar32)0x0130 : u_toupper(srcCodepoint));
-                U16_APPEND(lpDst, dstIdx, cwDstLength, dstCodepoint, isError);
-                assert(isError == FALSE && srcIdx == dstIdx);
-            }
-        }
-        else
-        {
-            // bToUpper && !bTurkishCasing
-            while (srcIdx < cwSrcLength)
-            {
-                U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
-                dstCodepoint = u_toupper(srcCodepoint);
-                U16_APPEND(lpDst, dstIdx, cwDstLength, dstCodepoint, isError);
-                assert(isError == FALSE && srcIdx == dstIdx);
-            }
-        }
-    }
-    else
-    {
-        if (bTurkishCasing)
-        {
-            // !bToUpper && bTurkishCasing
-            while (srcIdx < cwSrcLength)
-            {
-                // In turkish casing, LATIN CAPITAL LETTER I (U+0049) lower cases to
-                // LATIN SMALL LETTER DOTLESS I (U+0131).
-                U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
-                dstCodepoint = ((srcCodepoint == (UChar32)0x0049) ? (UChar32)0x0131 : u_tolower(srcCodepoint));
-                U16_APPEND(lpDst, dstIdx, cwDstLength, dstCodepoint, isError);
-                assert(isError == FALSE && srcIdx == dstIdx);
-            }
-        }
-        else
-        {
-            // !bToUpper && !bTurkishCasing
-            while (srcIdx < cwSrcLength)
-            {
-                U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
-                dstCodepoint = u_tolower(srcCodepoint);
-                U16_APPEND(lpDst, dstIdx, cwDstLength, dstCodepoint, isError);
-                assert(isError == FALSE && srcIdx == dstIdx);
-            }
-        }
-    }
+	if (bToUpper)
+	{
+		while (srcIdx < cwSrcLength)
+		{
+			// In turkish casing, LATIN SMALL LETTER I (U+0069) upper cases to LATIN
+			// CAPITAL LETTER I WITH DOT ABOVE (U+0130).
+			U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
+			dstCodepoint = ((srcCodepoint == (UChar32)0x0069) ? (UChar32)0x0130 : u_toupper(srcCodepoint));
+			U16_APPEND(lpDst, dstIdx, cwDstLength, dstCodepoint, isError);
+			assert(isError == FALSE && srcIdx == dstIdx);
+		}
+	}
+	else
+	{
+		while (srcIdx < cwSrcLength)
+		{
+			// In turkish casing, LATIN CAPITAL LETTER I (U+0049) lower cases to
+			// LATIN SMALL LETTER DOTLESS I (U+0131).
+			U16_NEXT(lpSrc, srcIdx, cwSrcLength, srcCodepoint);
+			dstCodepoint = ((srcCodepoint == (UChar32)0x0049) ? (UChar32)0x0131 : u_tolower(srcCodepoint));
```

Comment:
does Linux map correctly the Turkish uppercase i (0x0130)? or we needed to special case this one too?